name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/arkeadirectbank/ArkeaDirectBankPDFExtractorTest.java

```diff
@@ -254,6 +254,37 @@ public void testCompteAchat06()
                         hasTaxes("EUR", 0.00), hasFees("EUR", 0.00))));
     }
 
+    @Test
+    public void testCompteAchat07()
+    {
+        ArkeaDirectBankPDFExtractor extractor = new ArkeaDirectBankPDFExtractor(new Client());
+
+        List<Exception> errors = new ArrayList<>();
+
+        List<Item> results = extractor.extract(PDFInputFile.loadTestCase(getClass(), "Achat07.txt", "Taxes07.txt"),
+                        errors);
+
+        assertThat(errors, empty());
+        assertThat(countSecurities(results), is(1L));
+        assertThat(countBuySell(results), is(1L));
+        assertThat(countAccountTransactions(results), is(0L));
+        assertThat(results.size(), is(2));
+        new AssertImportActions().check(results, "EUR");
+
+        // check security
+        assertThat(results, hasItem(security( //
+                        hasIsin("FR0000124141"), hasWkn(null), hasTicker(null), //
+                        hasName("VEOLIA ENVIRONNEMENT"), //
+                        hasCurrencyCode("EUR"))));
+
+        // check buy sell transaction
+        assertThat(results, hasItem(purchase( //
+                        hasDate("2020-04-22T10:42:24"), hasShares(100), //
+                        hasSource("Achat07.txt; Taxes07.txt"), //
+                        hasAmount("EUR", 1774.2), hasGrossValue("EUR", 1765), //
+                        hasTaxes("EUR", 5.30), hasFees("EUR", 3.90))));
+    }
```

Comment:
Hello @mikaelpeltier 
thx for this pull request.

Please add the follow tests and rename tests and files...
Add `hasNote("xxxx"), //`

- Single import of `testCompteAchat07()`  <--- without TaxesTreatment07.txt
- Single import of `testTaxestreatment07()`  <--- without Achat07 and rename to TaxesTreatment07
- Together import of `testAchat07WithTaxesTreatment07()` <--- TaxesTreatment07.txt and Achat07.txt
- Together import of `Achat07WithTaxesTreatment07_SourceFilesReversed()` <--- Achat07.txt and TaxesTreatment07.txtAchat07.txt 

The rest looks very good.
Do you have any dividend transactions with taxes treatment???

Regards
Alex

Reply:
Thanks for your modifications, I didn't had time to do them yesterday.
It seems that I do not have dividend transactions with taxes treatment, in fact, they are free of taxes.